modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/validation/ParameterModelValidator.java

```diff
@@ -107,13 +110,13 @@ private void validateParameter(ParameterModel parameterModel, String ownerName,
     private void validateNameCollisionWithSubtypes(String ownerName, String ownerModelType, String extensionName, List<String> parameterNames, ParameterModel parameterModel)
     {
         Optional<MetadataType> subTypeWithNameCollision = subTypesMapping.getSubTypes(parameterModel.getType()).stream()
-                .filter(subtype -> parameterNames.contains(hyphenize(IntrospectionUtils.getAliasName(subtype)))).findFirst();
+                .filter(subtype -> parameterNames.contains(hyphenize(MuleExtensionAnnotationParser.getAliasName(subtype)))).findFirst();
 
         if (subTypeWithNameCollision.isPresent())
         {
             throw new IllegalParameterModelDefinitionException(
                     String.format("The parameter [%s] in the %s [%s] from the extension [%s] can't have the same name as the ClassName or Alias of the declared subType [%s] for parameter [%s]",
-                                  IntrospectionUtils.getAliasName(subTypeWithNameCollision.get()), ownerModelType, ownerName, extensionName,
+                                  MuleExtensionAnnotationParser.getAliasName(subTypeWithNameCollision.get()), ownerModelType, ownerName, extensionName,
```

Comment:
this is not what I meant with my comment. MuleExtensionAnnotationParser should not be used outside of the context of the AnnotationBasedDescriber. IntrospectionUtils should be used in any other context. MuleExtensionAnnotationParser can eventually have some alias related method which delegates into it